Statement "for" in Statement translator extended for attribute access.

indazoo opened the pull request:
Hi
I tried to convert some python code and it had statements like this:
`for self.lineIndex in xrange(self.lineIndex, len(self.lines))`
This threw an exception and the conversion process stopped because `self.lineIndex` was not supported in the `for` statement.
So I added this case for a smooth run in the "for" translation code. The output seems to be right I compared a few code segments between the two languages. But I'm new here in this project and may easily have overlooked something or it could be simplified or better modularised.

Additional note:
There were also problems with not included operators "%=" "\\\\" and "\\\\=" which are also in my fork but not in this branch I request here to pull into your project. I can create a pull request for these also if you wish. But honestly I did this so the App does not fail. Probably the code conversion is not 100% correct because c# has no direct equivalent for the python "\\\\" operator as it seems.

Commit:
Statement "for" in Statement translator extended for attribute access.

uxmal commented:
Could you please add a unit test for your new code in `StatementTranslator.Tests.cs`? You could take one of the existing `foreach` translations, copy it, and adapt it to your new code. Also, are you sure the resulting code is valid? As far as I can tell, C# doesn't permit anything but a variable as the iteration variable: see https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/statements#the-foreach-statement for details.

You may need to change the code so that the resulting code looks like
```C#
foreach (var _tmp_1 in Enumerable.Range(this.lineIndex, this.lines.Count)) {
   self.lineIndex = _tmp_1;
  /* body of loop*/
}
```

indazoo commented:
I'll try to add a unit test. 
To be honest, I quickly changed the code so it runs through for my specific python code.

The thing with the iteration variable is correct. But man...10'000 errors shown by VS. There are a lot of things which should be different. My code change was intended to let someone use your software. But I understand one can see things differently. In the coding community, long hours have been spent discussing throwing exceptions or not.

Commit:
Added correct access of class property in foreach loop.
Added test for this case (Stmt_foreach_class_property)

indazoo commented:
OK, took a deeper dive and fixed it as you wish. Added the intermediate variable and the assignment to the class property. Also a test case is now included. I did not run the test code, just used my own file containing the same python code.
In Git, I added the new changes to the same branch on my fork. I don't know how to restart the checks of AppVeyor and the checks for conflicts with the base branch.
Shall we close this pull request and I start a new one?

uxmal commented:
No need to start a new PR -- just push out to the same branch you made the PR from

uxmal commented:
I made a small change to your code, otherwise it looks great. Thanks for the PR, and I appreciate your patience with my insistence on unit tests; I find them to be invaluable not only for testing my code as I develop it but also to guarantee no code regressions occur.
